tests/lapi/lib.lua

```diff
@@ -0,0 +1,52 @@
+--[[
+SPDX-License-Identifier: ISC
+Copyright (c) 2023-2025, Sergey Bronnikov.
+
+Test helpers.
+]]
+
+-- The function determines a Lua version. The function does not
+-- use any standard library function or any global variable,
+-- it does not rely on the name of "_ENV" upvalue (as it can be
+-- renamed when Lua is being built).
+-- The source is [1], see also [2].
+--
+-- 1. http://lua-users.org/lists/lua-l/2016-05/msg00297.html
+-- 2. https://github.com/Kong/version.lua/blob/master/src/version.lua
+local function version()
+    local f, t = function()
+        return function() end
+    end, {
+        nil, -- luacheck: ignore
+        [false]  = "Lua 5.1",
+        [true]   = "Lua 5.2",
+        [1/'-0'] = "Lua 5.3",
+        [1]      = "LuaJIT",
+    }
+    return t[1] or t[1/0] or t[f()==f()]
+end
+
+local function bitwise_op(op)
+    return function(x, y)
+        local op_name = op
+        local expr = y and
+                     ("return %d %s %d"):format(x, op_name, y) or
+                     ("return ~%d"):format(x)
+        local bc = load(expr)
+        return bc and bc() or nil
```

Comment:
Looks like it is better to use `assert()` check here.

Reply:
Fixed:

```diff
-local function bitwise_op(op)
+local function bitwise_op(op_name)
     return function(x, y)
-        local op_name = op
         local expr = y and
                      ("return %d %s %d"):format(x, op_name, y) or
-                     ("return ~%d"):format(x)
-        local bc = load(expr)
-        return bc and bc() or nil
+                     ("return %s%d"):format(op_name, x)
+        return assert(load(expr))()
     end
 end
 
```